mapbox_vector_tile/decoder.py

```diff
@@ -89,9 +90,18 @@ def parse_geometry(self, geom, ftype):
             i = i + 1
 
             if cmd == CMD_SEG_END:
-                break
+                if ftype == POLYGON and coords:
+                    coords.append(coords[0])
+                parts.append(coords)
+                coords = []
+
+            elif cmd == CMD_MOVE_TO or cmd == CMD_LINE_TO:
+
+                if ftype == LINESTRING and coords and cmd == CMD_MOVE_TO:
```

Comment:
[The spec](https://github.com/mapbox/vector-tile-spec/pull/36/files#diff-3594e5e9052be0e33421df52071394d9R29) says:

> For polygons a repeated `move_to` SHALL indicate either another exterior of a new polygon part or an interior ring of the previous polygon part.

So if we're going to follow that, then we need to handle the case where `CMD_MOVE_TO` starts a new exterior or interior ring.